src/tests/multihost/ipa/test_adtrust.py

```diff
@@ -607,3 +607,77 @@ def test_ad_user_ssh_ipa_client(multihost):
                                         f'/var/log/sssd/sssd_{domain_name}.log')
+
+    @staticmethod
+    def test_skip_members_in_view_search(multihost):
+        """
+        :title: Skip group members that point to an entry in the views base search
+        :id: 40adfa6a-b951-4aab-aa5b-57e149526e8d
+        :customerscenario: true
+        :bugzilla:
+          https://bugzilla.redhat.com/show_bug.cgi?id=2151403
+        :description: When resolving the nested groups, any entry in 
+         cn=views,cn=accounts,$BASEDN (or whatever the user configured
+         using ipa_views_search_base) must be ignored.
+        :setup:
+         1. Configure trust between IPA server and AD.
+         2. Configure client machine with SSSD integrated to IPA.
+        :steps:
+          1. Add a user1 ID override in 'default trust view'.
+          2. Create ext_group as a external group and add user in admins and external group.
+          3. Make ext_group a member of the admins group.
+          4. Run an "id" command for the override user.
+        :expectedresults:
+          1. Successfully added user1 in default trust view.
+          2. Successfully created and added user in admins and external group.
+          3. Made external_group a member of the admins group.
+          4. Id command succeeds, admin group is visible, groups are properly resolved.
+        """
+        kinit_admin = 'kinit admin'
+        multihost.master[0].run_command(kinit_admin, stdin_text='Secret123',
+                                        raiseonerr=False)
+
+        domain = multihost.ad[0].domainname
+
+        ipa_client = sssdTools(multihost.client[0])
+        ipa_client.clear_sssd_cache()
+
+        # Here user1 is already added in AD and also member of group1, group2 and group3
+        # Add a new User ID override in default trust view.
+        create_newuser_override = f"ipa idoverrideuser-add 'default trust view' user1@{domain}"
+        multihost.master[0].run_command(create_newuser_override, raiseonerr=False)
+
+        # Add user to ‘admins’ group
+        add_user_override = f"ipa group-add-member admins --idoverrideusers user1@{domain}"
+        multihost.master[0].run_command(add_user_override, raiseonerr=False)
+
+        # Add a external group named as a ext_group
+        create_ext_grp = f'ipa group-add --external ext_group'
+        multihost.master[0].run_command(create_ext_grp, raiseonerr=False)
+
+        # Add a external group named as a ext_group
+        add_grp_members = f'ipa -n group-add-member ext_group --external=user1@{domain}'
+        multihost.master[0].run_command(add_grp_members, raiseonerr=False)
+
+        # Make ext_group a member of the admins group, as a result the members of ext_group will be
+        # members of admin too.
+        add_ext_grp_to_admin = f'ipa group-add-member admins --groups=ext_group'
+        multihost.master[0].run_command(add_ext_grp_to_admin, raiseonerr=False)
+
+        ipa_client.clear_sssd_cache()
```

Comment:
Hi,

I think it might be worth the invalidate or flush the SSSD cache on the master as well and check on the master that the user belongs to the `admins` group. Otherwise the result on the client might depend on the state of SSSD's caches on the master.

bye,
Sumit

Reply:
Hello,

@sumit-bose thanks for reviewing the code.
I have updated the code please check.

Reply:
Hi,

thanks, to be on the safe side I'd like to suggest to do everything on the master first, since the client is getting its data from the master. So, first clear the cache on master and then on the client, now lookup the user on master. Ideally you should check the result of the lookup on master now but I understand that you want to run the cleanup/teardown before the asserts. So lookup the user on the client, to the cleanup and then check the results from the ipa-server first to indicate if something is already wrong on the server or if only the client failed.

bye,
Sumit

Reply:
Update the code.

code will first check lookup on master and then lookup from master.
please check again.